apollo-ios/Sources/Apollo/ApolloClient.swift

```diff
@@ -1,51 +1,72 @@
-import Foundation
 import Dispatch
+import Foundation
+
 #if !COCOAPODS
-import ApolloAPI
+  import ApolloAPI
 #endif
 
 /// A cache policy that specifies whether results should be fetched from the server or loaded from the local cache.
-#warning("TODO: rethink this API. Onlye valid for queries currently")
+#warning("TODO: rethink this API. Only valid for queries currently")
 public enum CachePolicy: Sendable, Hashable {
   /// Return data from the cache if available, else fetch results from the server.
-  case returnCacheDataElseFetch
-  ///  Always fetch results from the server.
-  case fetchIgnoringCacheData
-  ///  Always fetch results from the server, and don't store these in the cache.
-  case fetchIgnoringCacheCompletely
-  /// Return data from the cache if available, else return an error.
-  case returnCacheDataDontFetch
+  case cacheFirst
+  /// Attempt to fetch results from the server, if failed, return data from the cache if available.
+  case networkFirst
+  ///  Fetch results from the server, do not attempt to read data from the cache.
+  case networkOnly
+
   /// Return data from the cache if available, and always fetch results from the server.
-  case returnCacheDataAndFetch
+  case cacheAndNetwork  // seperate function?
+
+  /// Return data from the cache if available, else return an error.
+  case cacheOnly  // replace with separate function?
 
-#warning("TODO: this unsafe is not properly made atomic. Fix this")
-  /// The current default cache policy.
-  nonisolated(unsafe) public static var `default`: CachePolicy = .returnCacheDataElseFetch
+  ///  Always fetch results from the server, and don't store these in the cache.
+  //  case fetchIgnoringCacheCompletely
+
+  #warning("TODO: this unsafe is not properly made atomic. Fix this")
+  //  /// The current default cache policy.
+  //  nonisolated(unsafe) public static var `default`: CachePolicy = .cacheFirst
```

Comment:
All changes in `ApolloClient` itself are just for POC and are still in flux.

The plan is to only include those first three cache policies.

Reply:
New CachePolicy API is implemented in #684